modules/core/shared/src/test/scala/gem/util/InstantMicrosSpec.scala

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package util
+
+import gem.arb.ArbTime._
+
+import cats.kernel.laws._
+import cats.tests.CatsSuite
+
+
+@SuppressWarnings(Array("org.wartremover.warts.Equals", "org.wartremover.warts.NonUnitStatements"))
+final class InstantMicrosSpec extends CatsSuite {
+
+  // Laws
+  checkAll("InstantMicro", OrderLaws[InstantMicros].order)
+
+  test("Construction should truncate Instant nanoseconds to microseconds") {
+    forAll { (i: InstantMicros) =>
```

Comment:
Should we test the round-trip?

Reply:
They are round-tripped in the `EphemerisDaoSpec` because they serve as `Ephemeris` keys.  An arbitrary `InstantMicros` may have any number of microseconds and the must be faithfully written and read from the database to pass the tests.

Reply:
👍